src/test/fuzz/pow.cpp

```diff
@@ -57,7 +57,7 @@ FUZZ_TARGET_INIT(pow, initialize_pow)
             } else {
                 current_block.nChainWork = ConsumeArithUInt256(fuzzed_data_provider);
             }
-            blocks.push_back(current_block);
+            blocks.push_back(&current_block);
```

Comment:
Pretty sure this will later-on result in a use-after-free?

Reply:
Argh, good catch. Thought I'd checked for that. Fixed.

Reply:
Hmm, the CI fuzz run initially passed on this test: https://cirrus-ci.com/task/5646660507271168?logs=ci#L3420 . I wonder why it didn't catch that.

Reply:
`valgrind` doesn't catch this either. Looks like the compiler will just optimize this UB away and set all objects to the same pointer, even if they are different?

This is mildly shocking, as it makes it impossible to catch this kind of bug with any tool.

Reply:
That's scary. Glad my fuzz-via-idiocy technique is yielding more novel info about C++ compilers.